### PR TITLE
Align community contact email validation across backend and frontend

### LINKED ISSUE #738: Align community contact email validation across backend and frontend
## Problem

Community contact email addresses are validated differently across system boundaries:

- The backend uses Pydantic `EmailStr`.
- The frontend response parser applies a separate ASCII-oriented regular expression before rendering a `mailto:` link.

An address accepted and stored by the backend can therefore cause the entire community-events query to fail frontend validation.

## Relevant code

- `backend/app/schemas.py` community edition contact fields
- `frontend/src/components/CommunityEvents.tsx` response validation
- `frontend/src/config/constants.ts` email regular expression

## Expected behavior

The backend and frontend should share a compatible validation contract. The frontend must still prevent unsafe `mailto:` values, including control/header-injection characters, without rejecting legitimate API values unexpectedly.

## Acceptance criteria

- [ ] Define the canonical accepted email-address contract
- [ ] Frontend validation accepts every address the backend can legitimately return, or the backend intentionally enforces the same documented subset
- [ ] Unsafe control characters and `mailto:` header injection remain rejected
- [ ] Tests cover internationalized/edge-case valid addresses and malicious values
- [ ] A malformed contact address does not unnecessarily hide unrelated community events

### PR DESCRIPTION
**Pull request by tjorim**
## Summary

Closes #738.

Community contact email addresses were validated differently on each side of
the API boundary: the backend used Pydantic's `EmailStr` (which, by default,
accepts internationalized/Unicode addresses), while the frontend applied a
much narrower, ASCII-only regex before rendering a `mailto:` link. An address
the backend accepted and stored could fail frontend validation — and because
that failure was a thrown error inside the shared response parser, it took
down the *entire* community events section (all edition types), not just the
one bad edition.

- **Canonical contract**: added a shared, documented ASCII "RFC 5321 dot-atom"
  email pattern — `COMMUNITY_CONTACT_EMAIL_PATTERN` in
  `backend/app/schemas.py` and the matching `COMMUNITY_CONTACT_EMAIL_REGEX` in
  `frontend/src/config/constants.ts`. It's a superset of the old frontend
  regex's character set (covers the full RFC 5321 atext local-part charset)
  while still rejecting control characters and other `mailto:`
  header-injection vectors.
- **Backend enforcement**: `EditionCreate`/`EditionUpdate` now validate
  `external_contact_email` against this pattern on write, intentionally
  rejecting internationalized/Unicode addresses (they can't safely round-trip
  through a `mailto:` link) rather than relying on `EmailStr`'s more
  permissive default behavior.
- **Frontend resilience**: `CommunityEvents.tsx`'s response parser no longer
  throws on a malformed `external_contact_email`. It drops just that field
  (logging a warning) so the edition and its events — and any unrelated
  editions — still render normally.

## Test plan

- [x] `backend`: `uv run pytest tests/test_editions.py -q` — new parametrized
      coverage for accepted addresses (RFC 5321 atext local parts, dotted
      local parts) and rejected ones (`mailto:` query/header injection,
      control characters, unquoted Unicode local part/domain, and a
      punycode-encoded domain that Pydantic normalizes back to Unicode)
- [x] `backend`: `uv run pytest -q` (full suite, 367 passed)
- [x] `backend`: `uv run ruff check app tests`, `uv run ty check app`
- [x] `frontend`: `vitest run tests/components/CommunityEvents.test.tsx` —
      updated/added coverage confirming a bad contact email is dropped
      without hiding the edition, a control-character injection attempt is
      dropped, a legitimate edge-case address is accepted and rendered as a
      safe `mailto:` link, and a malformed email on one edition type doesn't
      hide a sibling edition type
- [x] `frontend`: full `vitest run` (432 passed), `pnpm typecheck`, `pnpm lint`

---
_Generated by [Claude Code](https://claude.ai/code/session_01UbKMxQbFDbzcPXXwYAp2Q6)_

### CODE REVIEW

File: frontend/src/config/constants.ts [hunk 1]
```diff
@@ -15,3 +15,14 @@ export const CAROUSEL_AUTOPLAY_DELAY_MS = 3000;
...
+/**
+ * Canonical contract for community edition contact emails, shared with the
+ * backend's COMMUNITY_CONTACT_EMAIL_PATTERN (backend/app/schemas.py). ASCII-only
+ * RFC 5321 "dot-atom" local part plus a conventional domain — a superset of
+ * EMAIL_REGEX's character set that still excludes control/whitespace characters,
+ * so a matching value is always safe to render as a `mailto:` link. Keep both
+ * patterns in sync.
+ */
+export const COMMUNITY_CONTACT_EMAIL_REGEX =
+  /^[A-Za-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[A-Za-z0-9!#$%&'*+/=?^_`{|}~-]+)*@[A-Za-z0-9](?:[A-Za-z0-9-]{0,61}[A-Za-z0-9])?(?:\.[A-Za-z0-9](?:[A-Za-z0-9-]{0,61}[A-Za-z0-9])?)*\.[A-Za-z]{2,}$/;
```
[the remark this answers is not in the document]
Comment (tjorim) [reply frontend/src/config/constants.ts:28]: Not a bug: `/` doesn't need escaping inside a `[...]` character class in a JS regex literal — the literal delimiter is only significant outside character classes. `/^[/]/.test("/")` compiles and returns `true` in Node, and this exact regex passed `oxlint`, `tsc`, and the vitest suite (including a test asserting a match) before this PR was opened. Leaving as-is.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: frontend/src/components/CommunityEvents.tsx [hunk 2]
```diff
@@ -74,14 +74,20 @@ function readOptionalString(
...
 ): string | undefined {
   const value = readOptionalString(record, key, context);
-  if (value !== undefined && !EMAIL_REGEX.test(value)) {
-    throw new Error(`${context}.${key} must be a valid email address or null.`);
+  if (value !== undefined && !COMMUNITY_CONTACT_EMAIL_REGEX.test(value)) {
+    console.warn(`${context}.${key} is not a valid, safe email address; omitting it.`);
+    return undefined;
   }
   return value;
 }
```
[the remark this answers is not in the document]
Comment (tjorim) [reply frontend/src/components/CommunityEvents.tsx:96]: Good catch — fixed in 76a90e6, checking the type directly instead of delegating to `readOptionalString`, plus a regression test for a non-string `external_contact_email`.

---
_Generated by [Claude Code](https://claude.ai/code)_
